### PR TITLE
Milestone 7 : Write Unit test (Pair programming)

### PR DESCRIPTION
**Pull request by Chrissiku**
## Project requirements
- Add unit tests to your classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/teacher_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'spec_helper'
+
+describe Teacher do
+  # The teacher should inherit from Person
+  it 'Inherit from Person' do
+    expect(Teacher.superclass).to eq Person
+  end
+
+  # Instantiate the Teacher class.
```
Comment (BenMukebo) [inline spec/teacher_spec.rb:9]: - [ ] Great job so far 👏🏻 👏🏻 everything look fine congrats 🥳  Just one thing miss, according to the project requirements, there should be a unit tests for all the methods in all the classes, Kindly make sure the `can_use_services?` class method has (at least) an 'it' block 👇🏻 check your teacher,rb file

```ruby
  # Method can_use_services?
  def can_use_services?
    true
  end
```

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+describe Student do
+  # The student should inherit from Person
+  it 'Inherit from Person' do
+    expect(Student.superclass).to eq Person
+  end
+
+  # Instantiate the Student class.
+  before :each do
+    @student = Student.new 20, 2, 'chris'
+  end
```
Comment (BenMukebo) [inline spec/student_spec.rb:12]: - [ ] Same here. According to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `play_hooky` class method has (at least) an 'it' block 👇🏻 check your student,rb file


```ruby
 # Method play_hooky
  def play_hooky
    "¯\(ツ)/¯"
  end
```

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @Chrissiku

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻

- No linter errors. ✔️ 
- Gitflow was correctly used.  ✔️   
- Professional README ✔️ 
- There are unit tests for methods in classes. ✔️
- All test files are placed in a different directory. ✔️
- There is one file per class. ✔️ 
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/student_spec.rb (hunk 2 above)
Comment (Chrissiku) [reply spec/student_spec.rb:12]: You are right, Let's implement it as soon as possible

**Review by BenMukebo — APPROVED** [review]
### Hi @Chrissiku , 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 




## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Chrissiku** [comment]
Thank you

**Pull request merged**